app/main.py

```diff
@@ -1,3 +1,57 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+from decimal import Decimal
+
+from app.customer import Customer, Car
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as file:
+        json_data = json.load(file)
+    Car.change_fuel_price(json_data["FUEL_PRICE"])
+    customers = [
+        Customer(
+            customer["name"],
+            customer["product_cart"],
+            customer["location"],
+            customer["money"],
+            customer["car"]
+        )
+        for customer in json_data["customers"]
+    ]
+    shops = [
+        Shop(
+            shop["name"],
+            shop["location"],
+            shop["products"]
+        )
+        for shop in json_data["shops"]
+    ]
+    for customer in customers:
+        purchase_flag = False
+        print(f"{customer.name} has {customer.money} dollars")
+        min_cost = Decimal("infinity")
+        shop_index = 0
+        for index, shop in enumerate(shops):
+            all_cost = (
+                customer.car.calculate_cost_distance(customer, shop)
+                + shop.calculate_purchase(customer)
+            )
+            print(
+                f"{customer.name}'s trip to the "
+                f"{shop.name} costs {all_cost}"
+            )
+            if all_cost < min_cost:
+                min_cost = all_cost
+                shop_index = index
+        if min_cost <= customer.money:
+            purchase_flag = True
+        if purchase_flag:
+            print(customer.arrive_at_shop(shops[shop_index]))
```

Comment:
i think all this prints and function calls would rather be put in Customer class as a method because all of this actions can be done within Customer instance - you can call it `do_shopping` or something.  Having so many prints inside your main() makes it look messy - it's better to organize this logic